src/components/CytoscapeGraph/CytoscapeTrace.ts

```diff
@@ -27,7 +27,25 @@ const showSpanSubtrace = (cy: Cy.Core, graphType: GraphType, span: Span) => {
   // From upstream to downstream: Parent app or workload, Inbound Service Entry, Service, App or Workload, Outbound Service Entry
   let lastSelection: Cy.NodeCollection | undefined = undefined;
 
-  if (graphType === GraphType.APP) {
+  if (graphType === GraphType.SERVICE) {
+    // In service graph type, parent can be a Service or a Workload (e.g. when it initiates the transaction)
+    const sourceAppNs = searchParentApp(span);
+    if (sourceAppNs) {
+      let selector = `node[!${CyNode.isGroup}][${CyNode.nodeType}="${NodeType.SERVICE}"][${CyNode.app}="${sourceAppNs.app}"][${CyNode.namespace}="${sourceAppNs.namespace}"]`;
+      let parent = cy.elements(selector);
+      if (!parent || parent.length === 0) {
+        // Try workload
+        const sourceWlNs = searchParentWorkload(span);
+        if (sourceWlNs) {
+          selector = `node[${CyNode.workload}="${sourceWlNs.workload}"][${CyNode.namespace}="${sourceWlNs.namespace}"]`;
+          parent = cy.elements(selector);
+        }
```

Comment:
What do you think about this and avoid another if confition?

```ts
const sourceWlNs = searchParentWorkload(span);
if ((!parent || parent.length === 0) && sourceWlNs) {
        // Try workload
          selector = `node[${CyNode.workload}="${sourceWlNs.workload}"][${CyNode.namespace}="${sourceWlNs.namespace}"]`;
          parent = cy.elements(selector);
}
```
do you make sense to you?

Reply:
Note: I didn't check searchParentWorkload(span) , maybe it's a lot of complexity in that method

Reply:
Thanks for the review.
`searchParentWorkload` is neither very complex nor very trivial, but still, I'd prefer not to do unnecessary call here.

Reply:
understood so we can merge this one

Reply:
post-merge review :-)  The logic looks OK to me, I think it should work as expected.  Style-wise I may have used a switch statement but that's just personal pref.

Reply:
OMG Sorry my fault @jshaughn :'( I didn't see the others reviewers assign :(

Reply:
Not a problem, @aljesusg.   I usually wait for the PR developer to perform the merge, but it's not a strict process on our team.  Anyway, no big deal!